src/main/scala/tech/sourced/gemini/Gemini.scala

```diff
@@ -494,5 +502,36 @@ object Gemini {
       RepoFile(row.getString(meta.repo), row.getString(meta.commit), row.getString(meta.path), row.getString(meta.sha))
     }
   }
+
+  def getCommunities(conn: Session,
+                     keyspace: String,
+                     communities: List[(Int, List[Int])],
+                     elementIds: Map[String, Int]): Iterable[Iterable[RepoFile]] = {
+
+    val idToSha1 = for ((elem, id) <- elementIds) yield (id, elem.split("@")(1))
+
+    // Transform communities of element IDs to communities of sha1s
+    // Equivalent to apollo graph.py BatchedCommunityResolver._gen_hashes
+    // https://github.com/src-d/apollo/blob/f51c5a92c24cbedd54b9b30bab02f03e51fd27b3/apollo/graph.py#L295
+    val communitiesSha1 = communities.map {
+      case (communityId, community) => community.collect {
+        case id if (idToSha1.contains(id)) => idToSha1(id)
+      }
+    }.filter(_.size > 1)
```

Comment:
It took me a while to parse nested patten matching.

Would this be equivalent of

```scala
    val communitiesSha1 = communities
      .map { case (_, community) =>
        community
          .filter(idToSha1.contains)
          .map(idToSha1)
      }
      .filter(_.size > 1)
```

?